propsigner/propsigner.go

```diff
@@ -0,0 +1,142 @@
+package propsigner
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	pb "github.com/textileio/go-auctions-client/gen/wallet"
+	logger "github.com/textileio/go-log/v2"
+)
+
+const (
+	v1Protocol            = "/auctions/proposal-signer/1.0.0"
+	maxRequestMessageSize = 100 << 10 // 100KiB
+
+	// filecoinDealProtocolV1 refers exactly to the libp2p protocol used to send deal proposals.
+	// This value is used to know how to unmarshal proposal payloads. Future deal proposal versions
+	// might be supported, and thus we need to be able to distinguish them to do proper unmarshaling.
+	filecoinDealProtocolV1 = "/fil/storage/mk/1.1.0"
+)
+
+var (
+	log            = logger.Logger("propsigner")
+	streamDeadline = time.Minute
+
+	errInvalidAuthToken  = errors.New("invalid auth token")
+	errWalletMissingKeys = errors.New("wallet doesn't have keys for address")
+)
+
+// Wallet contains private keys for Filecoin addresses.
+type Wallet interface {
+	Has(addr string) (bool, error)
+	Sign(addr string, payload []byte) (*crypto.Signature, error)
+}
```

Comment:
At the libp2p protocol level, we rely on an abstract `Wallet` to do signing.

For `auc wallet daemon` that would mean the explained `localwallet.Wallet` that contains the provided keys and use `go-filsigner`.
For people using this as a library, they can plug in whatever they want. Maybe doing RPC calls to a real Lotus API to do signing, or just whatever satisfies the interface.